forage-android/src/main/java/com/joinforage/forage/android/ecom/services/vault/bt/BTResponseParser.kt

```diff
@@ -15,48 +18,61 @@ internal class BTResponseParser(btRes: Result<Any?>) : VaultResponseParser {
 
     val isSuccessful: Boolean
 
+    private val failureResponseRegExp = BtFailureResponseRegExp(btRes)
+
     init {
         // BT returns a Result<Any?> so it's never null
         vaultError = parseVaultError(btRes)
-        forageError = parseForageError(BaseResponseRegExp(btRes))
+        forageError = parseForageError(failureResponseRegExp)
         isSuccessful = btRes.isSuccess && vaultError == null && forageError == null
         successfulResponse = parseSuccessfulResponse(btRes)
     }
 
     private fun parseVaultError(vaultResponse: BasisTheoryResponse): ForageApiResponse.Failure? {
         if (vaultResponse.isSuccess) return null
-        val resRegExp = BtResponseRegExp(vaultResponse)
-        return try {
-            // In AbstractVaultSubmitter we track the forageError
-            // and the forage status code of the UnknownErrorApiResponse
-            // via the VaultProxyResponseMonitor. This keeps us informed
-            // of *when* erroroneous BT response happen. Unfortunately,
-            // we do not currently track the specifics of the proxy_error
-            // that that BT returned to us.
-            // TODO: log the specific error that BT responds with when
-            //  resRegExp.containsProxyError == true
-            if (resRegExp.containsProxyError) UnknownErrorApiResponse else null
-        } catch (_: Exception) {
-            null
-        }
+        // In AbstractVaultSubmitter we track the forageError
+        // and the forage status code of the UnknownErrorApiResponse
+        // via the VaultProxyResponseMonitor. This keeps us informed
+        // of *when* erroroneous BT response happen. Unfortunately,
+        // we do not currently track the specifics of the proxy_error
+        // that that BT returned to us.
+        // TODO: log the specific error that BT responds with when
+        //  resRegExp.containsProxyError == true
+        return if (failureResponseRegExp.containsProxyError) UnknownErrorApiResponse else null
     }
 
-    private fun parseForageError(resRegExp: BaseResponseRegExp): ForageApiResponse.Failure? {
-        if (resRegExp.bodyText == null || resRegExp.statusCode == null) return null
-        return try {
-            ForageApiResponse.Failure(resRegExp.statusCode, resRegExp.bodyText)
-        } catch (_: Exception) {
+    private fun parseForageError(resRegExp: BtFailureResponseRegExp): ForageApiResponse.Failure? =
+        if (resRegExp.bodyText == null ||
+            resRegExp.statusCode == null ||
+            // if there's a proxy_error, don't try to parse as a Forage error
+            resRegExp.containsProxyError
+        ) {
             null
+        } else {
+            // if there's a body, a status code, and no proxy_error, then
+            // we embrace that the following line will throw an exception
```

Comment:
This exception get's caught in `BtPinSubmitter`, though. It is returned as a `UnknownServerError`